### PR TITLE
feature/ryanbuui/875877867-add-create-agency-btn)

### PR DESCRIPTION
**Pull request by ryanbuui**
### Administrative Info

Monday Board ID: 851212620
Make sure your branch name conforms to: `<feature/staging/hotfix/...>/[username]/[Monday Item ID]-[3-4 word description separated by dashes]`. Otherwise, please rename your branch and create a new PR.

### Changes

What changes did you make?

- Added and styled CreateAgencyBtn component
- Updated navbar route with AgencyProfileForm
- Modified route structure

### Testing

How did you confirm your changes worked?

- Created and ran tests using react-testing-library and jest

### Confirmation of Change

Upload a screenshot, if possible. Otherwise, please provide instructions on how to see the change.

CreateAgencyBtnPic
NavbarAgencyProfileFormPic

### CODE REVIEW

File: __tests__/CreateAgencyBtn.test.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    ...jest.requireActual('react-router-dom'),
+    useHistory: () => ({
+        push: mockHistoryPush,
+    }),
+}));
+
+// Checks if the rendered button from the CreateAgencyBtn component redirects to the correct URL
+describe('CreateAgencyBtn', () => {
+    it('Button goes to the correct URL on click', () => {
+      const { getByRole } = render(
```
Comment (wllmwu) [inline __tests__/CreateAgencyBtn.test.js:19]: It looks like your Javascript has 4-space indents but your JSX has 2-space indents. I think we should make them match—Emmanuel and I used 2 spaces for all the form components

File: src/App.js [hunk 2]
```diff
@@ -1,11 +1,27 @@
 import React from 'react';
 import Navbar from "./components/Navbar/Navbar";
 import './App.css';
+import { BrowserRouter } from 'react-router-dom';
+import { BrowserRouter as Router, Route, Switch, Link } from "react-router-dom"
+import Home from "./pages/Home/Home";
+import Agency from "./pages/Agency/Agency";
+import AgencyProfileForm from "./pages/agency-profile-form/AgencyProfileForm";
```
Comment (wllmwu) [inline src/App.js:8]: Nick's PR will change the directory structure of our project, so sometime in the near future we will have to standardize all the file paths and update imports. As far as I can tell this PR should be ok though

File: src/components/Navbar/Navbar.js [hunk 3]
```diff
@@ -24,19 +20,9 @@ class Navbar extends Component {
...
                             )
-
                         })}
                     </ul>
                 </nav>
-                <div>
-                    <switch>
-                        <div>
-                            <Route exact path="/" component={Home}/>
-                            <Route exact path="/agency" component={Agency}/>
```
Comment (wllmwu) [inline src/components/Navbar/Navbar.js:35]: How nitpicky are our style requirements? In this file, sometimes there are spaces around equals signs in DOM elements and sometimes there aren't, and there aren't spaces before the slashes here

File: src/pages/Agency/Agency.js [hunk 4]
```diff
@@ -1,10 +1,12 @@
 import React, { Component } from 'react'
+import CreateAgencyBtn from "../../components/CreateAgencyBtn/CreateAgencyBtn";
 
 export default class Agency extends Component {
```
Comment (wllmwu) [inline src/pages/Agency/Agency.js:4]: I think Nick suggested keeping `export` statements at the end of files, so we should stick to that

File: src/pages/Agency/Agency.js [hunk 5]
```diff
@@ -1,10 +1,12 @@
 import React, { Component } from 'react'
+import CreateAgencyBtn from "../../components/CreateAgencyBtn/CreateAgencyBtn";
```
Comment (wllmwu) [inline src/pages/Agency/Agency.js:2]: One import statement has a semicolon and one doesn't :(

or should i say ;(

**Review by wllmwu — CHANGES_REQUESTED** [review]
My only complaints are with code style—some indentation, spacing, line enders, etc. are inconsistent and I guess we need to pass a code review. If you're using VS Code, the Prettier extension can automatically format an entire file on save which is super helpful

File: __tests__/CreateAgencyBtn.test.js (hunk 1 above)
Comment (sblumin) [reply __tests__/CreateAgencyBtn.test.js:19]: Agreed, let's keep a consistency in indentation with 2 spaces = 1 tab

File: __tests__/navbar.test.js [hunk 6]
```diff
@@ -0,0 +1,27 @@
+import React from 'react';
+import { BrowserRouter } from "react-router-dom"; 
+import { BrowserRouter as Router, Route, Switch, Link } from "react-router-dom"
```
Comment (sblumin) [inline __tests__/navbar.test.js:3]: let's add semicolons to the end of all import statements to keep it consistent!

File: src/App.js (hunk 2 above)
Comment (sblumin) [reply src/App.js:8]: Make sure to add a semicolon to import statement on line 5 as well

File: src/components/CreateAgencyBtn/CreateAgencyBtn.css [hunk 7]
```diff
@@ -0,0 +1,24 @@
+button {
+  border: none;
+  background: none;
+  display: flex;
+  justify-content: space-between;
+}
+.res-circle {
```
Comment (sblumin) [inline src/components/CreateAgencyBtn/CreateAgencyBtn.css:7]: Can you add whitespaces between each of these style elements? And also potentially inline comments for each or a file header to just clarify what the styling is for

File: src/components/CreateAgencyBtn/CreateAgencyBtn.js [hunk 8]
```diff
@@ -0,0 +1,18 @@
+import React from 'react';
+import { useHistory } from "react-router-dom";
+import './CreateAgencyBtn.css'
```
Comment (sblumin) [inline src/components/CreateAgencyBtn/CreateAgencyBtn.js:3]: add a semicolon here

File: src/components/Navbar/Navbar.js (hunk 3 above)
Comment (sblumin) [reply src/components/Navbar/Navbar.js:35]: let's keep spaces consistent so update these to all follow the same style

**Review by sblumin — CHANGES_REQUESTED** [review]
Looks really good! I made some comments on small style changes that should be made so just address those!

File: __tests__/CreateAgencyBtn.test.js (hunk 1 above)
Comment (ryanbuui) [reply __tests__/CreateAgencyBtn.test.js:19]: Yea, sorry about that, styling slipped our minds for some of the parts. I'll implement the changes sometime today and make sure everything is consistent.

**Comment by ryanbuui** [comment]
Sorry for the delay, got caught up in some class work. I applied all the styling changes you guys asked for and also made some additional edits to maintain consistency. If you have any other feedback, don't hesitate to many any requests! Thanks again.

**Review by sblumin — APPROVED** [review]
LGTM! Thanks for addressing those changes!

**Review by wllmwu — APPROVED** [review]
Looks LGTM to me

**Pull request merged**